### PR TITLE
Move SCFToCFG into Util dialect and later in the pipeline

### PR DESCRIPTION
**Pull request by rsuderman**
It is difficult to investigate buffer / dispatch behavior across CFG, while SCF is much simpler to trace their usage. Pushed SCF to CFG later in the pipeline.

### CODE REVIEW

File: compiler/src/iree/compiler/InputConversion/StableHLO/Passes.cpp [hunk 1]
```diff
@@ -46,9 +46,9 @@ void buildStableHLOInputConversionPassPipelineImpl(
       stablehlo::createLegalizeControlFlow());
 
   passManager.addPass(createFlattenTuplesInSCF());
-  if (detuple) {
-    passManager.addPass(createFlattenTuplesInCFG());
-  }
+  // if (detuple) {
+  passManager.addPass(createFlattenTuplesInCFG());
+  // }
```
Comment (benvanik) [inline compiler/src/iree/compiler/InputConversion/StableHLO/Passes.cpp]: ?

**Review by benvanik — CHANGES_REQUESTED** [review]
let's put this in flow for now instead - I'd like to be able to support scf all the way down and not need this pass at all so I want to avoid it being invoked elsewhere further down/in other places (more likely when it's in a common util/ location and easier for people to grab)

File: compiler/src/iree/compiler/InputConversion/StableHLO/Passes.cpp (hunk 1 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/InputConversion/StableHLO/Passes.cpp]: Blah I'll fix that. I am lazy about finding the right stablehlo input conversion.

**Comment by rsuderman** [comment]
> let's put this in flow for now instead - I'd like to be able to support scf all the way down and not need this pass at all so I want to avoid it being invoked elsewhere further down/in other places (more likely when it's in a common util/ location and easier for people to grab)

Moved. Hopefully we can just strip it sometime in the future.

File: compiler/src/iree/compiler/Dialect/Util/Transforms/PassDetail.h [hunk 2]
```diff
@@ -8,6 +8,7 @@
 #define IREE_COMPILER_DIALECT_UTIL_TRANSFORMS_PASS_DETAIL_H_
 
 #include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/FunctionInterfaces.h"
```
Comment (benvanik) [inline compiler/src/iree/compiler/Dialect/Util/Transforms/PassDetail.h]: can revert the Util/ changes

**Review by benvanik — APPROVED** [review]

File: compiler/src/iree/compiler/Dialect/Util/Transforms/PassDetail.h (hunk 2 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/Dialect/Util/Transforms/PassDetail.h]: Done.

**Pull request merged**